types/wlr_keyboard.c

```diff
@@ -194,26 +190,13 @@ void wlr_keyboard_set_keymap(struct wlr_keyboard *kb,
 		kb->mod_indexes[i] = xkb_map_mod_get_index(kb->keymap, mod_names[i]);
 	}
 
-	keymap_str = xkb_keymap_get_as_string(kb->keymap,
+	kb->keymap_string = xkb_keymap_get_as_string(kb->keymap,
```

Comment:
The previous `kb->keymap_string` should be free'd before replacing it with the new one. Preferably it should be replaced only if `xkb_keymap_get_as_string` doesn't fail.